### PR TITLE
[RW-6445][risk=low] Dev tool for uploading new snapshots of Terra Methods to Agora (WDLs) 

### PR DESCRIPTION
**Pull request by ericsong**
Description:
Two usages
All environments
`./project.rb create-terra-method-snapshot --source-git-repo broadinstitute/gatk --source-git-path scripts/variantstore/wdl/ngs_cohort_extract.wdl --source-git-branch ah_var_store --method-name AgoraDevToolTest1 --all-projects`

Single environment
`./project.rb create-terra-method-snapshot --source-git-repo broadinstitute/gatk --source-git-path scripts/variantstore/wdl/ngs_cohort_extract.wdl --source-git-branch ah_var_store --method-name AgoraDevToolTest2 --project all-of-us-workbench-test`

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/tools/src/main/java/org/pmiops/workbench/tools/CreateTerraMethodSnapshot.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+                .get();
+
+        try {
+          methodResponse =
+              apiClientFactory
+                  .methodRepositoryApi()
+                  .createMethodSnapshot(
+                      methodNamespace,
+                      methodName,
+                      Integer.toString(latestSnapshotId),
```
Comment (calbach) [inline api/tools/src/main/java/org/pmiops/workbench/tools/CreateTerraMethodSnapshot.java:137]: Hm.. this is confusing. This is the snapshot ID of the most recent, not the one that's being created right? What would be the significance of using an older snapshot ID here?

File: api/libproject/devstart.rb [hunk 2]
```diff
@@ -1998,6 +1997,83 @@ def create_wgs_cohort_extraction_bp_workspace(cmd_name, *args)
...
+  op.parse.validate
+
+  projects = op.opts.all_projects ? ENVIRONMENTS.keys - ["local"] : [op.opts.project]
+
+  projects.each { |project|
+    flags = ([
+      ["--config-json", get_config_file(project)],
+      ["--source-git-repo", op.opts.source_git_repo],
+      ["--source-git-path", op.opts.source_git_path],
+      ["--source-git-ref", op.opts.source_git_ref],
```
Comment (calbach) [inline api/libproject/devstart.rb:2056]: One thing that's not being captured here is the commit hash. Also there's a small concern that a commit could go in between when `staging` and `prod` are getting applied, and we wouldn't easily see a difference.

One possible solution is to resolve branch ref into a sha1 (before the project loop) before passing off the value to the CLI.

File: api/tools/src/main/java/org/pmiops/workbench/tools/CreateTerraMethodSnapshot.java [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+import org.pmiops.workbench.firecloud.model.FirecloudMethodResponse;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Create new Terra method snapshot in Agora
+ *
+ * <p>Assumptions - The provided Agora method namespace does not exist yet OR the cohort extraction
```
Comment (calbach) [inline api/tools/src/main/java/org/pmiops/workbench/tools/CreateTerraMethodSnapshot.java:32]: I don't see code to create the namespace - does it just get lazily created when you create a method if it doesn't exist yet?

**Review by ericsong — COMMENTED** [review]
thanks for the review @calbach  should be good for round 2

File: api/libproject/devstart.rb [hunk 4]
```diff
@@ -1998,6 +1999,88 @@ def create_wgs_cohort_extraction_bp_workspace(cmd_name, *args)
...
+  op.opts.source_git_repo = "broadinstitute/gatk"
+  op.add_option(
+    "--source-git-repo [source-git-repo]",
+    ->(opts, v) { opts.source_git_repo = v},
+    "git owner/repo where the source file is located. default: broadinstitute/gatk")
+  op.opts.source_git_path = "scripts/variantstore/wdl/ngs_cohort_extract.wdl"
+  op.add_option(
+    "--source-git-path [source-git-path]",
+    ->(opts, v) { opts.source_git_path = v},
+    "git path where the source file is located, relative to the repo's root directory. default: scripts/variantstore/wdl/ngs_cohort_extract.wdl")
```
Comment (calbach) [inline api/libproject/devstart.rb:2031]: nit: should probably be possible to inline the above variable here

**Review by calbach — APPROVED** [review]

**Comment by calbach** [comment]
Also, validateSwagger looks to have legitimately failed, which may be a first. Appears to be a real issue in the firecloud yaml

**Pull request merged**